xbmc/view/GUIViewState.cpp

```diff
@@ -101,7 +101,17 @@ CGUIViewState* CGUIViewState::GetViewState(int windowId, const CFileItemList& it
     return new CGUIViewStateLibrary(items);
 
   if (items.IsPlayList())
-    return new CGUIViewStateMusicPlaylist(items);
+  {
+    // Playlists (like .strm) can be music or video type
+    // we set the playlist type based on the current
+    // GUI NAV window, otherwise fallback to general
+    if (windowId == WINDOW_VIDEO_NAV)
+      return new CGUIViewStateVideoPlaylist(items);
+    else if (windowId == WINDOW_MUSIC_NAV)
+      return new CGUIViewStateMusicPlaylist(items);
+    else
+      return new CGUIViewStateGeneral(items);
```

Comment:
why general? shouldn't it be `return new CGUIViewStateMusicPlaylist(items);` just like before?

Reply:
yes changed